### DISCUSSION TITLE
How to access multi language links?

### DISCUSSION THREAD

**Discussion by richi** [datenstrom/community#152] [Ask a question] [answered]
We are working on a multi language site and we need language specific links to the search pages. In every language folder we have one (hidden) "search" folder. English is the default language and therefore the link to the english search is "/search/". To figure out this link on an english page as well as (for example) a "/fr/search/" link on a "/fr/blog/" page we use the following code:

``` PHP
$searchLocation = "/search/";
$pagesMultiLanguage = $yellow->pages->multi($searchLocation, false, true); 
foreach($pagesMultiLanguage as $page)
{
    if ($page->get("language") == $yellow->page->get("language"))
    {
        $searchLocation = $page->getLocation();
    }
}
```

This works, but looks a little verbose to me. I guess there is an easier way to fight the "/xx/search/" versus "/search/" logic. Any Ideas?

**Comment by markseuffert** [comment]
Interesting problem, I looked around if there's an internal function that could be used. I couldn't find a single line to create a multi language location. How about this alternative: 

``` php
$homeLocation = $yellow->pages->getHomeLocation($yellow->page->location);
$searchLocation = rtrim($homeLocation, '/')."/search/";
```

Looks like an easy way is missing in the [Yellow API](https://datenstrom.se/yellow/help/api-for-developers). Anyone else has an idea?

**Comment by richi** [comment]
Quick & dirty :) Works well, but only as long as one does not need to know wether the "/xx/page" actually exists.

**Comment by markseuffert** [comment]
Here's another alternative, If you don't know if a language specific page exists:

``` php
$searchLocation = "/search/";
$homeLocation = $yellow->pages->getHomeLocation($yellow->page->location);
$page = $yellow->pages->find(rtrim($homeLocation, '/').$searchLocation);
if($page) $searchLocation = $page->location;
```

You could also make a search for all visible pages on your website, not just the current language. If you want to give it a try, replace `index(false, false)` with `index(false, true)` in the [search plugin](https://github.com/annaesvensson/yellow-search).

**Comment by richi** [comment, marked as answer]
Just in case somebody else stumbles over the same problem: To get the multi language navigation to work with "hidden" pages one has to change _multi($yellow->page->location, false, **true**)_ accordingly. 

Maybe this _**true**_ should be the default in _navigation-multilanguage.php_?

**Comment by markseuffert** [comment]
Good idea! Switching between languages should work on any kind of page. Want to send a pull request?

**Comment by richi** [comment]
Here you go.

**Comment by markseuffert** [comment]
Thank you. 

Good luck with the multi language website :smile:
